File: 0560/solution.py
from collections import defaultdict


class Solution:
    def subarraySum(self, nums: list[int], k: int) -> int:
        prefix_sums = defaultdict(int)
        total = 0
        res = 0
        prefix_sums[0] = 1
        for num in nums:
            total += num
            res += prefix_sums[total - k]
            prefix_sums[total] += 1
        return res

File: 0560/solution.ts
function subarraySum(nums: number[], k: number): number {
  const sums: Map<number, number> = new Map();
  sums.set(0, 1);
  let res = 0;
  let curr = 0;
  for (let i = 0; i < nums.length; i++) {
    curr += nums[i];
    const diff = curr - k;
    res += sums.has(diff) ? sums.get(diff)! : 0;
    sums.set(curr, (sums.get(curr) || 0) + 1);
  }
  return res;
}
